packages/contracts-bedrock/test/OptimismPortal.t.sol

```diff
@@ -163,145 +163,258 @@ contract OptimismPortal_Test is Portal_Initializer {
     }
 
     /// @dev Tests that `depositTransaction` succeeds for an EOA depositing a tx with 0 value.
-    function test_depositTransaction_noValueEOA_succeeds() external {
-        // EOA emulation
-        vm.prank(address(this), address(this));
-        vm.expectEmit(true, true, false, true);
-        emitTransactionDeposited(
-            address(this), NON_ZERO_ADDRESS, ZERO_VALUE, ZERO_VALUE, NON_ZERO_GASLIMIT, false, NON_ZERO_DATA
+    function testFuzz_depositTransaction_noValueEOA_succeeds(
+        address _to,
+        uint64 _gasLimit,
+        bytes memory _data
+    )
+        external
+    {
+        _gasLimit = uint64(
+            bound(_gasLimit, op.minimumGasLimit(uint64(_data.length)), systemConfig.resourceConfig().maxResourceLimit)
         );
 
-        op.depositTransaction(NON_ZERO_ADDRESS, ZERO_VALUE, NON_ZERO_GASLIMIT, false, NON_ZERO_DATA);
+        // EOA emulation
+        vm.expectEmit(address(op));
+        emitTransactionDeposited({
+            _from: address(this),
+            _to: _to,
+            _value: 0,
+            _mint: 0,
+            _gasLimit: _gasLimit,
+            _isCreation: false,
+            _data: _data
+        });
+
+        vm.prank(address(this), address(this));
```

Comment:
The comment says "EOA emulation" but now the tx.origin has code since it's this test contract. Should we just define a new EOA with `address depositor = makeAddr("depositor")` and then `vm.prank(depositor, depositor)`?

Reply:
EOA detection in the portal uses `msg.sender == tx.origin`

Reply:
but yeah, it's a fair point that we could use a real EOA.

Reply:
Yep as-is this would pass the `msg.sender == tx.origin` but is not realistic since origin has code. I don't think it should matter, but probably safer to use a realistic scenario